pkg/plugins/aws/ami/condition.go

```diff
@@ -0,0 +1,74 @@
+package ami
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/scm"
+)
+
+// Condition test if a image matching specific filters exist.
+func (a *AMI) Condition(source string) (bool, error) {
+
+	if source != "" {
+		// Based on source information,
+		// we try to define a default image-id resource
+		// if not researched
+		isImageIDDefined := false
+		for i := 0; i < len(a.Filters); i++ {
+			if strings.Compare(a.Filters[i].Name, "image-id") == 0 {
+				isImageIDDefined = true
+				break
+			}
+
+		}
+
+		// Set image-id to source output if not yet defined
+		if !isImageIDDefined {
+			a.Filters = append(a.Filters, Filter{
+				Name:   "image-id",
+				Values: source,
+			})
+		}
+
+	} else {
+		if len(a.Filters) == 0 {
+			logrus.Infof("\u2717 No AMI could be found as no AMI filters defined\n")
+			return false, nil
+		}
+	}
+
+	svc, errs := a.Init()
+
+	for _, err := range errs {
+		logrus.Error(err)
+	}
+	if len(errs) > 0 {
+		return false, errors.New("Too many errors")
+	}
+
+	result, err := a.GetLatestAmiID(svc)
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(result) > 0 {
```

Comment:
Question: if there are no result found, should'nt the method raise an error?

Reply:
That's a good question that I asked myself a while ago and I decided that it wasn't an error because I want to know if an image id exist or not, To me an error could using a wrong endpoint